streams/src/main/java/org/apache/kafka/streams/kstream/KStreamBuilder.java

```diff
@@ -115,6 +117,13 @@ public KStreamBuilder() {
         return KStreamImpl.merge(this, streams);
     }
 
+    /**
+     * Create a unqiue processor name used for translation into the processor topology.
+     * This function should only be used for users to add their own DSL operators and corresponding translation rules.
```

Comment:
@ymatsuda I was thinking that developers could add their own operators with the translation rules, which hence needs to call this function. Maybe I was thinking too much as there won't be many people doing it :). May be just change to "This function is for internal usage only?".